src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpHandler.cs

```diff
@@ -1632,67 +1108,36 @@ private void SetRequestHandleTlsOptions(SafeWinHttpHandle requestHandle)
 
         private void SetRequestHandleClientCertificateOptions(SafeWinHttpHandle requestHandle, Uri requestUri)
         {
-            // Must be HTTPS scheme to use client certificates.
-            if (requestUri.Scheme != UriSchemeHttps)
+            if (requestUri.Scheme != WinHttpRequestState.UriSchemeHttps)
             {
                 return;
             }
 
-            // Get candidate list for client certificates.
-            X509Certificate2Collection certs;
+            X509Certificate2 clientCertificate = null;
             if (_clientCertificateOption == ClientCertificateOption.Manual)
             {
-                certs = ClientCertificates;
+                clientCertificate = WinHttpCertificateHelper.GetEligibleClientCertificate(ClientCertificates);
```

Comment:
This code matches Desktop behavior.

Valid client certs can be:
- Have no EKU attribute.  Therefore, all "uses" are valid including client cert
- Have an EKU attribute but missing the client OID.  This is NOT a valid client cert.
- Have an EKU attribute and contain the client OID.  This is a valid cert.